=== modified file 'vxi11_cmd.cc'
--- vxi11_cmd.cc	2009-01-27 18:50:18 +0000
+++ vxi11_cmd.cc	2015-06-10 14:28:02 +0000
@@ -23,14 +23,15 @@
  */
 
 #include "vxi11_user.h"
-#define BUF_LEN 100000
+#define BUF_LEN (1000*1024*1024)
+
+char		buf[BUF_LEN];
 
 int	main(int argc, char *argv[]) {
 
 static char	*device_ip;
 static char	*device_name;
 char		cmd[256];
-char		buf[BUF_LEN];
 int		ret;
 long		bytes_returned;
 CLINK		*clink;
@@ -68,9 +69,22 @@
 		if (strstr(cmd, "?") != 0) {
 			bytes_returned = vxi11_receive(clink, buf, BUF_LEN);
 			if (bytes_returned > 0) {
-				printf("%s\n",buf);
+				if ((device_name != NULL) && (strcmp(device_name,"Rigol_DS1054") == 0) && (strcmp(cmd,":DISPLAY:DATA?") == 0)) {
+				     /* print of binary data does not work with printf("%s\n"); */
+				     int i;
+				     /* print header normally */
+				     for (i = 0; i < 11; i++) {
+				       printf("%c",buf[i]);
+				     }
+				     /* print bmp file content in hex */
+				     for (i = 11; i < bytes_returned-11; i++) {
+				       printf("%02x",int(buf[i]) & 0xff);
+				     }
+				     printf("\n");
+			        } else {
+				      printf("%s\n",buf);
 				}
-			else if (bytes_returned == -15) {
+			} else if (bytes_returned == -15) {
 				printf("*** [ NOTHING RECEIVED ] ***\n");
 				}
 			else	break;

